### ISSUE TITLE
Versions chart broken if load balancer is used?

### ISSUE THREAD

**Issue by larsxschneider** [Autodesk/hubble#184]
The versions chart uses the `haproxy.log` to determine the Git client versions in use. This has two disadvantages:

- only HTTPS Git connections are monitored
- if a load balancer is used then the source IP is `127.0.0.1` in the haproxy log and therefore we cannot distinguish different machines

**Comment by larsxschneider** [comment]
This snippet should get the versions from the audit log:
```
$ zcat -f /var/log/github-audit.log.1* | perl -ne 'print if s/.*agent=git\/(\d+(?:\.\d+){0,2}).*"user_id":(\d+).*/\2\t\1/' | sort | uniq | perl -lape 's/\d+ *//' | sort -r -V | uniq -ic
```

This should also include SSH connections. However, I wonder should take *any* Git client into considerations with a regex like this:
```
zcat -f /var/log/github-audit.log.1* | perl -ne 'print if s/.*agent=([^\/" ]+\/\d+(?:\.\d+){0,2}).*"user_id":(\d+).*/\2\t\1/' | sort | uniq | perl -lape 's/\d+ *//' | sort | uniq -ic
```

This would also record `JGit` clients and the output would look like this:
```
      1 	git/2.9.0
      2 	git/2.9.2
      3 	git/2.9.3
      4 	JGit/4.5.0
      5	JGit/4.6.1
      6 	JGit/4.8.0
```
@pluehne Could that be easily be made compatible with your versions check logic? Could everything not `git/*` be considered "unknown"?

**Comment by pluehne** [comment]
@larsxschneider: Yes, that would totally work, and I prefer this approach to the one we had before (not just because it counts the Git version distribution more accurately, but also because of the fact that we can record the used clients in this way).

We’d have to think whether we want to support a mixed data format in `git-versions.tsv` or rather have a migration transform the old data into the new format. Probably, the former would be easier to implement.
